### ISSUE TITLE
When running Update-AzVm with a VM that was created from an Image, a UserData error sometimes appears unexpectedly

### ISSUE THREAD

**Issue by Sandido** [Azure/azure-powershell#16689]
### Description

When running Update-AzVm with a VM that was made from an existing image, a UserData error may occur stating:
"UserData must be in Base64 encoding and with a maximum length of 87384 characters." This occurred in our PS test framework when I was re-recording old tests. 

This appears due to an unexpected string being piped into the Update-AzVm Userdata parameter, the string being "Microsoft.Azure.Commands.Compute.Models.PSVirtualMachine" causing the UserData ValidateBase64Encoded logic to fire and see this string as being not base64 encoded. 
This occurs because the Userdata string parameter has the piping metadata tag ValureFromPipeline.

Solution:
Changing ValueFromPipeline to ValueFromPipelineByPropertyName seems to resolve the issue. 

I am submitting this bug report so there is a public record of its detection and resolution.

### Issue script & Debug output

```PowerShell
$loc = "eastus";
New-AzResourceGroup -Name $rgname -Location $loc -Force;

        # VM Profile & Hardware
        $vmsize = 'Standard_DS2_v2';
        $vmname = 'vm' + $rgname;

        # NRP
        $subnet = New-AzVirtualNetworkSubnetConfig -Name ('subnet' + $rgname) -AddressPrefix "10.0.0.0/24";
        $vnet = New-AzVirtualNetwork -Force -Name ('vnet' + $rgname) -ResourceGroupName $rgname -Location $loc -AddressPrefix "10.0.0.0/16" -Subnet $subnet;
        $vnet = Get-AzVirtualNetwork -Name ('vnet' + $rgname) -ResourceGroupName $rgname;
        $subnetId = $vnet.Subnets[0].Id;
        $pubip = New-AzPublicIpAddress -Force -Name ('pubip' + $rgname) -ResourceGroupName $rgname -Location $loc `
                                       -AllocationMethod 'Static' -DomainNameLabel ('pubip' + $rgname) -Zone "1" -Sku 'Standard';
        $pubip = Get-AzPublicIpAddress -Name ('pubip' + $rgname) -ResourceGroupName $rgname;
        $pubipId = $pubip.Id;
        $nic = New-AzNetworkInterface -Force -Name ('nic' + $rgname) -ResourceGroupName $rgname -Location $loc -SubnetId $subnetId -PublicIpAddressId $pubip.Id;
        $nic = Get-AzNetworkInterface -Name ('nic' + $rgname) -ResourceGroupName $rgname;
        $nicId = $nic.Id;

        # Storage Account (SA)
        $stoname = 'sto' + $rgname;
        $stotype = 'Standard_GRS';
        New-AzStorageAccount -ResourceGroupName $rgname -Name $stoname -Location $loc -Type $stotype;
        $stoaccount = Get-AzStorageAccount -ResourceGroupName $rgname -Name $stoname;

        # OS & Image
        $user = "Foo12";
        $password = $PLACEHOLDER;
        $securePassword = ConvertTo-SecureString $password -AsPlainText -Force;
        $cred = New-Object System.Management.Automation.PSCredential ($user, $securePassword);
        $computerName = 'test';

        $p = New-AzVMConfig -VMName $vmname -VMSize $vmsize -Zone "1" `
             | Add-AzVMNetworkInterface -Id $nicId -Primary `
             | Set-AzVMOperatingSystem -Windows -ComputerName $computerName -Credential $cred;

        #adam $imgRef = Get-DefaultCRPImage -loc $loc;
        $imgRef = Create-ComputeVMImageObject -loc "eastus" -publisherName "MicrosoftWindowsServerHPCPack" -offer "WindowsServerHPCPack" -skus "2012R2" -version "4.5.5198";
        $p = $imgRef | Set-AzVMSourceImage -VM $p;
        New-AzVM -ResourceGroupName $rgname -Location $loc -VM $p;

        $vm = Get-AzVM -ResourceGroupName $rgname -Name $vmname;
        Assert-AreEqual "1" $vm.Zones;

        $vm | Update-AzVM;
        Assert-AreEqual "1" $vm.Zones;
```

### Environment data

```PowerShell
N/A
```

### Module versions

```PowerShell
Az.Compute 4.20.0 and up.
```

### Error output

```PowerShell
System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: UserData must be in Base64 encoding and with a maximum length of 87384 characters.
ErrorCode: InvalidParameter
ErrorMessage: UserData must be in Base64 encoding and with a maximum length of 87384 characters.
ErrorTarget: userData
StatusCode: 400
ReasonPhrase: Bad Request
```

**Comment by dingmeng-xue** [comment]
We will look into it. If it is blocking you, please let us know.

**Comment by Sandido** [comment]
@dingmeng-xue , it is blocking other work I have for re-recording PS tests. 
The Userdata parameter is something [I added recently](https://msazure.visualstudio.com/One/_workitems/edit/10250208). I used only the piping value of ValueFromPipeline because the core PS team had mentioned using ValueFromPipelineByPropertyName was not preferred. However, I then discovered this buggy behavior and raised this item.
